src/helpers/worker-hours-info.model.ts

```diff
@@ -240,9 +240,11 @@ export class WorkerHourInfo {
       if (!VerboseDateHelper.isWorkingDay(day)) {
         return calculateFreeHours;
       }
+      const shiftSubtraction = SHIFTS[actualShift].normSubtraction ?? WORK_HOURS_PER_DAY;
+
       const subtractFromNorm =
-        actualShift === historyShift
-          ? WORK_HOURS_PER_DAY
+        actualShift === historyShift || !ShiftHelper.isNotWorkingShift(historyShift)
```

Comment:
Bug, który znalazłem wynikał z tego warunku. Jeżeli zmieniślimy w grafiku aktualnym na inną zmianę niepracującą niż ta w bazowym to od normy nic nie odejmowaliśmy.